Runtime/Scripts/PersistentConnection.cs

```diff
@@ -131,192 +217,193 @@ IEnumerator OnIntervalEdgeEventsLocation(EdgeEventsConnection connection)
-              Logger.Log("Received Cloudlet Update, New CloserCloudlet found,New Cloudlet GPS :" + CloudletLocation.GpsLocationFieldNumber);
-              if (fcTriggers.Contains(FindCloudletEventTrigger.CloserCloudlet))
-              {
-                findCloudletEvent.trigger = FindCloudletEventTrigger.CloserCloudlet;
-                findCloudletEvent.newCloudlet = edgeEvent.NewCloudlet;
-                bool fcResult = await integration.FindCloudlet();
-                integration.NewFindCloudletHandler(EdgeEventsStatus.success, findCloudletEvent);
-              }
-              break;
-            case EventError:
-              Logger.Log("Received EventError, :" + edgeEvent.ErrorMsg);
-              FindCloudletEvent fcEvent = new FindCloudletEvent() { trigger = FindCloudletEventTrigger.Error };
-              integration.NewFindCloudletHandler(EdgeEventsStatus.error, fcEvent);
-              break;
-            default:
-            case EventUnknown:
-              Logger.Log("Received Unknown Event");
-              break;
+            PropagateError(FindCloudletEventTrigger.LatencyTooHigh,
+              "New Cloudlet obtained from FindCloudletPerformanceMode is the same as old cloudlet");
           }
         }
+        else
+        {
+          PropagateError(FindCloudletEventTrigger.LatencyTooHigh, "FindCloudletPerformanceMode failed");
+        }
+      }
+    }
 
-    private void OnApplicationFocus(bool focus)
+    async void CompareLatencies(Statistics receivedStats)
     {
-      if (locationUpdatesRunning)
+      AppPort appPort;
+      Site site;
+      var netTest = new NetTest(integration.matchingEngine);
+      if (hasTCPPorts)
+      {
+        appPort = integration.GetAppPort(LProto.Tcp, config.latencyTestPort == 0 ? 0 : config.latencyTestPort);
+        site = new Site { host = appPort.FqdnPrefix + integration.latestFindCloudletReply.Fqdn, port = appPort.PublicPort, testType = TestType.CONNECT };
+      }
+      else
       {
-        StartCoroutine(OnIntervalEdgeEventsLocation(integration.matchingEngine.EdgeEventsConnection));
+        appPort = integration.GetAppPort(LProto.Udp);
+        site = new Site { host = appPort.FqdnPrefix + integration.latestFindCloudletReply.Fqdn, port = appPort.PublicPort, testType = TestType.PING };
       }
-      if (latencyUpdatesRunning)
+      await netTest.TestSite(site);
+      double normalizedLatency = receivedStats.Avg - (receivedStats.Avg * config.performanceSwitchMargin);
+      if (site.average < normalizedLatency)
       {
-        StartCoroutine(OnIntervalEdgeEventsLatency(integration.matchingEngine.EdgeEventsConnection, integration.GetHost()));
+        PropagateSuccess(FindCloudletEventTrigger.LatencyTooHigh, integration.latestFindCloudletReply);
+      }
+      else
+      {
+        PropagateError(FindCloudletEventTrigger.LatencyTooHigh,
+          "Cloudlet obtained from FindCloudletPerformanceMode have higher latency than the previous cloudlet");
```

Comment:
nit: "has higher"

Reply:
nit: this comment should be more like: "Latency threshold exceeded, but no other cloudlets have a meaningful improvement in latency"